### PR TITLE
Fix `BoundaryAdherence` NaN handling

### LINKED ISSUE #470: Fix BoundaryAdherence NaN handling
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDMetrics version: 0.11.0 (latest)
* Python version: 3.10
* Operating System: MacOS

### Error Description
Currently, the `BoundaryAdherence` metric ignores NaN values when calculating the percentage of values that fall within the min/max range of the real values. NaN values in the synthetic data should only be ignored if the real data contains NaNs, otherwise they should be considered out of bounds.

### Proposed Fix
When computing, only drop NaN values from the synthetic data if there are NaN values present in the real data.

### Steps to reproduce
```python
import pandas as pd
import numpy as np

from sdmetrics.single_column import BoundaryAdherence

real_data = pd.Series([2.3, 4.2, 1.0, 10.0, 9.2])
synthetic_data = pd.Series([3.4, np.nan, 1.2, 4.9, 9.1])

BoundaryAdherence.compute(real_data, synthetic_data)
```

### PR DESCRIPTION
**Pull request by frances-h**
CU-86ayga3rk

Resolve #470

### CODE REVIEW

**Review by lajohn4747 — APPROVED** [review]
lgtm

File: tests/unit/single_column/statistical/test_boundary_adherence.py [hunk 1]
```diff
@@ -31,6 +33,65 @@ def test_compute(self):
...
+        out of bounds if the real data does not also containt nan values.
+        """
+        # Setup
+        real_data = pd.Series([
+            datetime(2020, 10, 1),
+            datetime(2021, 1, 2),
+            datetime(2021, 9, 12),
+            datetime(2022, 10, 1),
+
+        ], dtype='datetime64[ns]')  # 0.8 -> 2.6
```
Comment (R-Palazzo) [inline tests/unit/single_column/statistical/test_boundary_adherence.py:70]: What does indicate # 0.8 -> 2.6 in the datetime case?

**Review by R-Palazzo — APPROVED** [review]
LGTM!

File: tests/unit/single_column/statistical/test_boundary_adherence.py (hunk 1 above)
Comment (frances-h) [reply tests/unit/single_column/statistical/test_boundary_adherence.py]: Ah good catch, that was just to help me write the test 😅

**Pull request merged**